src/v/datalake/coordinator/tests/iceberg_file_committer_test.cc

```diff
@@ -111,6 +111,31 @@ class FileCommitterTest
         ASSERT_FALSE(res.has_error());
     }
 
+    void get_snap_data_files(
+      const iceberg::snapshot& snap, chunked_vector<ss::sstring>* uris) {
+        const auto& mlist_uri = snap.manifest_list_path;
+        auto mlist_res = manifest_io.download_manifest_list(mlist_uri).get();
+        ASSERT_TRUE(mlist_res.has_value());
+        const auto& mlist = mlist_res.value();
+
+        auto schema = datalake::default_schema();
+        auto pspec = iceberg::partition_spec::resolve(
+          datalake::hour_partition_spec(), schema.schema_struct);
+        ASSERT_TRUE(pspec.has_value());
```

Comment:
does ASSERT_TRUE work as expected in helper functions like this outside the context of the TEST macro?

Reply:
ISTR Nicolae had also made a change to make this more palatable without ASSERT_NO_FATAL_FAILURE(), but this test uses the macro anyway out of habit